tests/test_optimizer.py

```diff
@@ -1378,25 +1378,31 @@ def gen_expr(depth: int) -> exp.Expression:
         self.assertEqual(18, normalization_distance(gen_expr(3), max_=100))
         self.assertEqual(110, normalization_distance(gen_expr(10), max_=100))
 
-    def test_custom_annotators(self):
+    def test_spark_annotators(self):
+        """Test Spark annotators, mainly built-in string/binary functions"""
+
+        schema = {"tbl": {"bin_col": "BINARY", "str_col": "STRING"}}
+
+        def _assert_func_return_type(func: str, dialect: str, target_type: str):
+            ast = parse_one(f"SELECT {func} FROM tbl", read=dialect)
+            optimized = optimizer.optimize(ast, schema=schema, dialect=dialect)
```

Comment:
Let's only use `annotate_types` for this, running all optimizer rules is too much. You can slightly rewrite the query so that the columns are annotated and there's no need to even use `qualify`.